Lib/getpass.py

```diff
@@ -68,12 +71,37 @@ def unix_getpass(prompt='Password: ', stream=None):
                 old = termios.tcgetattr(fd)     # a copy to save
                 new = old[:]
                 new[3] &= ~termios.ECHO  # 3 == 'lflags'
+                if echochar:
+                    new[3] &= ~termios.ICANON
                 tcsetattr_flags = termios.TCSAFLUSH
                 if hasattr(termios, 'TCSASOFT'):
                     tcsetattr_flags |= termios.TCSASOFT
                 try:
                     termios.tcsetattr(fd, tcsetattr_flags, new)
-                    passwd = _raw_input(prompt, stream, input=input)
+                    if not echochar:
+                        passwd = _raw_input(prompt, stream, input=input)
+                        stream.write('\n')
+                        return passwd
+
+                    passwd = ""
```

Comment:
If the rest fails, we still need to fallback to somehow `passwd = fallback_getpass(prompt, stream)`. I think? Also, the function becomes more and more unreadable with too many nested levels, so maybe we could split this part into a separate function?

Reply:
I'm all for separating this part into a separate function, but what should we call it, maybe `_echochar_input(prompt, stream, input)`?

regarding fallback, we still have calls to `fallback_getpass` in except blocks. I didn't remove it

Reply:
Yes, but the problem is that we have the check `if passwd is not None` which would be true when setting `passwd = ""`. So I don't know what do to in this case (if we hit the exception here, this means we were writing the password ... so we cannot just switch the input method =/)

Reply:
As for how to call it, maybe `_input_with_echochar`? or `_input_with_feedback`

Reply:
I'll change to `_input_with_echochar`